comms/core/src/connection_manager/peer_connection.rs

```diff
@@ -318,11 +336,43 @@ impl PeerConnection {
             .map_err(|_| PeerConnectionError::InternalReplyCancelled)?
     }
 
-    pub(crate) async fn disconnect_silent(&mut self, minimized: Minimized) -> Result<(), PeerConnectionError> {
+    /// Request to disconnect the peer connection if unused by other services.
+    pub async fn disconnect_if_unused(
+        &mut self,
+        minimized: Minimized,
+        requester: &str,
+    ) -> Result<(), PeerConnectionError> {
+        let number_of_rpc_clients = self.number_of_rpc_clients.load(Ordering::Relaxed);
+        let substream_count = self.substream_count();
+        if number_of_rpc_clients >= 1 || substream_count >= 1 {
```

Comment:
```suggestion
        if number_of_rpc_clients > 1 || substream_count > 1 {
```
Should this not be > 1?
If we try and disconnect and we where the only caller here, it would have 1 rpc client or substream?

Reply:
Here is an example log showing soft vs. hard disconnect decision - those counts go down to zero:
```
TRACE Soft disconnect - requester: 'Health check', peer: `d4b68aa41844d331149939e1da`, RPC clients: 0, substreams 2, NOT disconnecting
TRACE Hard disconnect - requester: 'Health check', peer: `d025bc9e4bd423a9b304c491b8`, RPC clients: 0, substreams 0
```

Reply:
And some more logs:
```
PeerConnection `1bf8c94450fd5f59650c9dc3d5` drop called, open sub-streams: 0, notified 1 potential RPC clients to drop connection
PeerConnection `d025bc9e4bd423a9b304c491b8` drop called, open sub-streams: 0, RPC clients: 0
PeerConnection `d7c289e9e3c8377705ce599a96` drop called, open sub-streams: 0, RPC clients: 0
PeerConnection `cb84ab7690c6ddd234ea7aea07` drop called, open sub-streams: 0, notified 1 potential RPC clients to drop connection
```